### PR TITLE
✨ testing wallet connect

### LINKED ISSUE #2476: Probably wrong ss58 on wallet connect
- Probably ss58 is handled wrong
- Missing identity, might be resolved by fixing the first one

image

Subwallet and Talisman spot another friction, probably link them together and tackle in one PR for best.

Related wallet issues
- #2480 
- #2477

### LINKED ISSUE #2477: Still not connect correctly when user install 2 wallets at the same time
### What happened?

I found it still not connect correctly when user install 2 wallets at the same time. pls check. 

The list of connected accounts shows only the first extension that connects to the Dapp. 

For example: when I connect to Polkadot JS and then connect to SubWallet, the list of accounts shows only Polkadot JS even if I have selected SubWallet and vice versa.

### Please reproduce in steps

Bug Detail: https://www.loom.com/share/752f5962149545679f0fbc315c153004

### Expected Behavior

It needs to show the correct address installed on different wallets even if the browser has multiple wallets installed at the same time. Like this our demo

### What browsers are you seeing the problem on?

Chrome

### Are you logged in?

Yes

### At which address did you encounter bug?

_No response_

### Screenshots

https://www.loom.com/share/3248348dafef4b28aee365b90e1e0b16

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #2484: Showing wrong wallet name in the alert when trying to connect to Subwallet
### What happened?

Showing wrong wallet name in the alert when trying to connect to Subwallet

### Please reproduce in steps

1. Prerequisite: Don't have Subwallet installed
2. Click `Connect` on the header
3. Click `Subwallet` in the popup
4. Look at the blue alert

### Expected Behavior

Current behavior showing: `You need to install the - talisman - browser extension`
Expected showing: `You need to install the - Subwallet - browser extension`

### What browsers are you seeing the problem on?

Chrome

### Are you logged in?

No

### At which address did you encounter bug?

nft.kodadot.xyz / kodadot.xyz

### Screenshots

typo

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #2500: Wallet selector should show icon of the wallet
### Is your feature request related to a problem?

I'm always frustrated when I click on the icon (like Talisman) and still see the KodaDot logo

### Describe the solution you would like

I would like to see the Talisman logo (or any selected wallet)

### Describe alternatives you have considered

I have tried this alternative

### Screeenshots

Screenshot 2022-03-05 at 10 31 11

Selected wallet Talisman 

Screenshot 2022-03-05 at 10 41 32

### PR DESCRIPTION
**Pull request by roiLeo**
## Connect Wallet 
Let's go!

### PR type

- [x] Bugfix

### What's new?

- [x] PR closes #2500  
- [x] PR closes #2484 
- [x] PR closes #2477 
- [x] PR closes #2476 
- [x] moar bugs to fix later

### Todo
- [ ] fix nova Wallet
- [x] fix change Account options
- [ ] extension approval on click
- [ ] store account in localstorage

### Screenrecord

https://user-images.githubusercontent.com/9987732/156402987-a0da0e6e-1aa1-46df-a61b-97359a04e33e.mov

Screenshot 2022-03-03 at 13-32-58 Junk #287

Screenshot 2022-03-07 at 09-40-01 NFT artist rank

### CODE REVIEW

**Comment by yangwao** [comment]
> Hieu Dao | SubWallet.App: error 2484 has been resolved but it seems that error 2477 is still unresolved
> Hieu Dao | SubWallet.App: https://github.com/kodadot/nft-gallery/issues/2477

File: components/common/WalletModal.vue [hunk 1]
```diff
@@ -61,10 +61,19 @@
 
         <div v-if="hasSelectedWalletProvider && hasWalletProviderExtension">
           <div class="subtitle has-text-centered">Choose your account</div>
-          <AccountSelect
-            v-model="account"
-            :label="$i18n.t('Account')"
-            :tooltip-visible="false" />
+
+          <b-field :label="$i18n.t('Account')">
```
Comment (vikiival) [inline components/common/WalletModal.vue:65]: Maybe move it to a separated component?

File: components/rmrk/Profile/NavbarProfileDropdown.vue [hunk 2]
```diff
@@ -64,9 +59,7 @@
       <b-dropdown-item custom aria-role="menuitem">
         <b-field>
           <p class="control">
-            <b-button type="is-primary" @click="changeAccount = !changeAccount">
-              Change account
-            </b-button>
+            <ConnectWalletButton buttonText="Change account" />
```
Comment (vikiival) [inline components/rmrk/Profile/NavbarProfileDropdown.vue:62]:
```suggestion
            <ConnectWalletButton buttonText="general.account.change" />
```

add translation pls 🥺

File: components/shared/AccountSelect.vue [hunk 3]
```diff
@@ -16,19 +16,14 @@
 <script lang="ts">
 import { Component, Prop, mixins } from 'nuxt-property-decorator'
 import WithKeyring, { KeyringAccount } from '@/utils/WithKeyring'
-import Tooltip from '@/components/shared/Tooltip.vue'
 
 @Component({
-  components: {
-    Tooltip,
-  },
+  components: {},
```
Comment (vikiival) [inline components/shared/AccountSelect.vue:21]:
```suggestion

```

File: components/shared/ConnectWalletButton.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <b-button type="is-primary" @click="openWalletConnectModal()">
+    {{ buttonText }}
```
Comment (vikiival) [inline components/shared/ConnectWalletButton.vue:3]:
```suggestion
    {{ $t(label) }}
```

**Comment by yangwao** [comment]
yah let's merge it )

File: components/shared/ConnectWalletButton.vue [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  </b-button>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'nuxt-property-decorator'
+import WalletModal from '~/components/common/WalletModal.vue'
+
+@Component({})
+export default class extends Vue {
+  @Prop({ default: 'Connect' }) public buttonText!: string // i18
```
Comment (vikiival) [inline components/shared/ConnectWalletButton.vue:13]:
```suggestion
  @Prop({ default: 'general.account.connetct' }) public label!: string // i18
```

File: utils/config/wallets/BaseDotsamaWallet.ts [hunk 6]
```diff
@@ -98,6 +99,7 @@ export class BaseDotsamaWallet implements Wallet {
     const unsubscribe = this._extension.accounts.subscribe(
       (accounts: InjectedAccount[]) => {
         const accountsWithWallet = accounts.map((account) => {
+          account.address = encodeAddress(decodeAddress(account.address), 2)
```
Comment (vikiival) [inline utils/config/wallets/BaseDotsamaWallet.ts:102]: I see some fixed ss58 prefix

Comment (vikiival) [reply utils/config/wallets/BaseDotsamaWallet.ts:102]: General there should be already wrapping function for this

**Review by vikiival — COMMENTED** [review]
Looks good, 
need some polish 💅

**Comment by vikiival** [comment]
Also app is blocked when until you don't accept/reject extensions
Screenshot 2022-03-05 at 10 29 09

**Comment by vikiival** [comment]
Also would change the design a bit (but that's a task for #2498) 
Screenshot 2022-03-05 at 10 31 11

File: utils/config/wallets/BaseDotsamaWallet.ts (hunk 6 above)
Comment (yangwao) [reply utils/config/wallets/BaseDotsamaWallet.ts:102]: yeah this won't work for statemine I guess

File: components/common/WalletModal.vue (hunk 1 above)
Comment (roiLeo) [reply components/common/WalletModal.vue:65]: actually that AccountSelect component, I've tried to edit it so I can pass accounts list as props but something wasn't working

File: utils/config/wallets/BaseDotsamaWallet.ts (hunk 6 above)
Comment (roiLeo) [reply utils/config/wallets/BaseDotsamaWallet.ts:102]: you find it, what should we do here? I've seen that we have many function to decode but I haven't found one that really works

**Comment by roiLeo** [comment]
> Also app is blocked when until you don't accept/reject extensions (#2480) Screenshot 2022-03-05 at 10 29 09

What's the solution here? should we ask for extension approval until we click on wallet?

**Comment by vikiival** [comment]
> What's the solution here? should we ask for extension approval until we click on wallet?

yup that would be a good solution

The edge case is that we should somehow cache the last logged wallet (e.g which extenstion and address you used)

**Comment by roiLeo** [comment]
> The edge case is that we should somehow cache the last logged wallet (e.g which extenstion and address you used)

yup, I thought about using localstorage

**Review by vikiival — APPROVED** [review]
Toward most advanced NFTs

**Pull request merged**